### PR TITLE
DP-774 - crash with proguard

### PR DESCRIPTION
**Pull request by yosriz**
repeat https://github.com/kinecosystem/kin-sdk-android/pull/56 for supporting migration module in kin2

### CODE REVIEW

**Review by drEfrati — APPROVED** [review]
👍

**Comment by OrenZak** [comment]
@yosriz CI failed, can you fix that before merging ?

**Comment by yosriz** [comment]
@OrenZak I'm not sure why it was even started, there's no real CI here, I never configured it (that's originated from stellar and it won't work).
I'll do manual tests on the upper level (kin-core)

**Comment by OrenZak** [comment]
> @OrenZak I'm not sure why it was even started, there's no real CI here, I never configured it (that's originated from stellar and it won't work).
> I'll do manual tests on the upper level (kin-core)

@yosriz 
np, did you update Gson lib? what's the current version you have? the latest one is `2.8.5`.
And I'll suggest you remove travis.yml so it won't confuse anyone.

File: android-stellar-sdk/proguard-rules.pro [hunk 1]
```diff
@@ -12,6 +12,14 @@
 # Gson specific classes
 -dontwarn sun.misc.**
 
+# Application classes that will be serialized/deserialized over Gson
+-keep class org.stellar.sdk.responses.** { <fields>; }
```
Comment (AliAsadi) [inline android-stellar-sdk/proguard-rules.pro:16]: Does it keep inner classes also?

Comment (yosriz) [reply android-stellar-sdk/proguard-rules.pro:16]: actually, I've noticed in tests that's the second new rule that does the trick (i.e. R8 is the blame) so I'm not sure,  I kept also this rule as to follow gson instructions (make sense to tell ProGuard to keep data classes fields)

**Comment by yosriz** [comment]
@OrenZak didn't touch gson, also in kin-sdk, wanted to focus only on what's needed to upgrade build tooling, also removed travis.

File: android-stellar-sdk/proguard-rules.pro (hunk 1 above)
Comment (AliAsadi) [reply android-stellar-sdk/proguard-rules.pro:16]: Cool, Thanks 👍🏻

**Review by AliAsadi — APPROVED** [review]
🏅

**Pull request merged**